sips/committee_consensus.md

```diff
@@ -0,0 +1,486 @@
+	Quorum          uint64
+	FeeRecipientAddress [20]byte   `ssz-size:"20"`
+	Graffiti            []byte     `ssz-size:"32"`
+}
+
+// Operator represents an SSV operator node that is part of a committee
+type Operator struct {
+	OperatorID        OperatorID
+	ClusterID         ssv.ClusterID
+	SSVOperatorPubKey []byte `ssz-size:"294"`
+	Quorum, PartialQuorum uint64
+	// All the members of the committee
+	Committee []*CommitteeMember `ssz-max:"13"`
+}
+
+// CommitteeMember represents all data in order to verify a committee member's identity
+type CommitteeMember struct {
+	OperatorID        OperatorID
+	SSVOperatorPubKey []byte
+}
+
+// Duty interface
+type Duty interface {
+	DutySlot() spec.Slot
+}
+
+// CommitteeDuty aggregates attesting and sync committee duties
+type CommitteeDuty struct {
+	Slot         spec.Slot
+	BeaconDuties []*BeaconDuty
+}
+
+// CommitteeRunner manages the duty cycle for a certain slot
+type CommitteeRunner interface {
+	// Start the duty lifecycle for the given slot. Emits a message.
+    StartDuty(duty Duty) error
+	// Processes cosensus message
+    ProcessConsensus(consensusMessage *qbft.Message) error
+	// Processes a post-consensus message
+    ProcessPostConsensus(msg PartialSignatureMessages) 
+}
+
+// Committee runner implements the old runner interface
+type CommitteeRunner struct {
+	// Important fields only
+	Shares          map[ValidatorPubkey]Share
+    Operator        Operator
+	QBFTController *qbft.Controller
+	BeaconNetwork  *types.BeaconNetwork
+}
+
+// BeaconVote is the consensus data
+type BeaconVote struct {
+    BlockRoot         phase0.Root
+    Source            phase0.Checkpoint
+    Target            phase0.Checkpoint
+}
+
+func ConstructAttestation(vote BeaconVote, duty AttesterDuty) Attestation {
+    bits := bitfield.New(duty.CommitteeLength)
+    bits.Set(duty.ValidatorCommitteeIndex)
+
+    return Attestation{
+        Slot: duty.Slot,
+        BlockRoot: vote.BlockRoot,
+        Source: vote.Source,
+        Target: vote.Target,
+        CommiteeIndex: duty.CommitteeIndex,
+        AggregationBits: bits,
+    }
+}
+
+func ConstructSyncCommittee(vote BeaconVote, duty AttesterDuty) SyncCommitteeMessage{
+    return SyncCommitteeMessage
+    {
+        Slot: duty.Slot,
+        BlockRoot: vote.BlockRoot,
+        ValidatorIndex: duty.ValidatorIndex,
+    }
+}
+```
+#### PartialSignatureMessages
+
+The current structure that we have in code can be kept with small changes.
+
+```go
+type PartialSignatureMessages struct {
+	Type     PartialSigMsgType
+	Slot     phase0.Slot
+	Messages []*PartialSignatureMessage
+}
+
+// PartialSignatureMessage is a msg for partial Beacon chain related signatures (like partial attestation, block, randao sigs)
+type PartialSignatureMessage struct {
+	PartialSignature Signature `ssz-size:"96"` // The Beacon chain partial Signature for a duty
+	SigningRoot      [32]byte  `ssz-size:"32"` // the root signed in PartialSignature
+	Signer         OperatorID
```

Comment:
i think this can be removed since it's already in the SignedSSVMessage, right?

if this would be a big change, then maybe we could just move this field up to the PartialSignatureMessages struct, just so it won't repeat for every signature

Reply:
I think we all want to remove it. I don't remember the reason now why it's still there
Was there any? @GalRogozinski

Reply:
Just so that old runners don't change to reduce the amount of changes and risk for bugs in a rushed release...
you guys want to change it now?

Reply:
IMO its best to change in this now than doing another fork for this

Reply:
Opened issue here: https://github.com/ssvlabs/ssv-spec/issues/422
we may do it as part of the next fork